tests/test_nuget.py

```diff
@@ -76,3 +76,8 @@ def test_less(self):
         self.check_order(self.assertLess, "1.0.0-pre", "1.0.0.1-alpha")
         self.check_order(self.assertLess, "1.0.0", "1.0.0.1-alpha")
         self.check_order(self.assertLess, "0.9.9.1", "1.0.0")
+
+    def test_as_set_value(self):
+        vers1 = NugetVersion.build_value("1.0.1")
+        vers2 = NugetVersion.build_value("1.0.1")
+        assert {vers1, vers2} == {vers1}
```

Comment:
You could test more simply with "is" to validate that the id are the same: `assert vers1 is vers2` which is the essence of a proper hash.
The set test is derived and not as explicit. You could also rename this to `test_NugetVersion_hash` .... but now you are really testing the semver library here IMHO with build_value... but what you want is really testing the NugetVersion instead or in addition.

1. add a test for the plain `semver` library to highlight the issues of not using the build segment in the hash.
2. add tests that create plain NugetVersion with build segments and various 4th segment and so on that really test the issue you found and check is on these object not on the "build_value" output.

otherwise I feel like you may not be testing what you need to test

Reply:
> You could test more simply with "is" to validate that the id are the same: `assert vers1 is vers2` which is the essence of a proper hash.

I don't think `is` will work here because `is` is used to compare the identity of the object [i.e. object's memory address] and two objects with the same hash can have different identities if they reside in different memory addresses.

```python
>>> from univers.versions import NugetVersion as NV
>>> t = NV.build_value("1.2.3+12")
>>> t.__hash__()
2747090334158322414
>>> l = NV.build_value("1.2.3+12")
>>> l.__hash__()
2747090334158322414
>>> t is l
False
>>> id(t)
4517202288
>>> id(l)
4519102112
>>> k = l
>>> id(k)
4519102112
>>> k is l
True
```
Yeah, it's is bit misleading I will add a separate test for both `python-semver` and plain text `NugetVersion`

Reply:
Good point on the hash vs. id.! You could also just compare calling `assert hash(v1) == hash(v2)`